### PR TITLE
types: update the ImageContentSource to match ImageContentSourcePolicy

### PR DESCRIPTION
**Pull request by abhinavdahiya**
The ImageContentSourcePolicy was updated https://github.com/openshift/api/commit/7076b89b1973b197f4aed994d7783624aecdc0f2

These changes api from  the sets of sources to a list source->mirrors.

/cc @umohnani8 @mrunalp @mtrmac @smarterclayton 
/assign @patrickdillon

### CODE REVIEW

File: pkg/asset/ignition/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -201,19 +201,14 @@ func (a *Bootstrap) getTemplateData(installConfig *types.InstallConfig, releaseI
...
-		}
-		for i := 0; i < len(group.Sources); i++ {
-			registry := sysregistriesv2.Registry{}
-			registry.Endpoint.Location = group.Sources[i]
-			registry.MirrorByDigestOnly = true
-			for j := 0; j < len(group.Sources); j++ {
-				registry.Mirrors = append(registry.Mirrors, sysregistriesv2.Endpoint{Location: group.Sources[j]})
-			}
-			registries = append(registries, registry)
+	for _, group := range mergedMirrorSets(imageSources) {
```
Comment (mtrmac) [inline pkg/asset/ignition/bootstrap/bootstrap.go]:
```go
if len(groups.Mirrors) == 0 {
    continue
}
```

**Comment by mtrmac** [comment]
LGTM.

File: pkg/asset/ignition/bootstrap/registries.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+func mergedMirrorSets(sources []types.ImageContentSource) []types.ImageContentSource {
+	sourceSet := make(map[string][]string)
+	mirrorSet := make(map[string]sets.String)
+	orderedSources := []string{}
+
+	for _, group := range sources {
+		if _, ok := sourceSet[group.Source]; !ok {
+			orderedSources = append(orderedSources, group.Source)
+			sourceSet[group.Source] = nil
```
Comment (patrickdillon) [inline pkg/asset/ignition/bootstrap/registries.go]: nit: you don't need to set nil here (the if statement just confirmed it was nil)

**Comment by umohnani8** [comment]
LGTM
/retest

**Comment by patrickdillon** [comment]
tested & /lgtm

File: pkg/asset/ignition/bootstrap/registries.go (hunk 2 above)
Comment (abhinavdahiya) [reply pkg/asset/ignition/bootstrap/registries.go]: @patrickdillon we need to set some value in the `map`. the if confirms no key exists for that source.

**Comment by abhinavdahiya** [comment]
> tested & /lgtm

needs to be in separate line

File: pkg/asset/ignition/bootstrap/registries.go (hunk 2 above)
Comment (patrickdillon) [reply pkg/asset/ignition/bootstrap/registries.go]: @abhinavdahiya  you can append to the key even if there is no value assigned: 

> Another instance of helpful zero values is a map of slices. Appending to a nil slice just allocates a new slice, so it's a one-liner to append a value to a map of slices; 
([source](https://blog.golang.org/go-maps-in-action))

Perhaps I am missing something else. Unfortunately we can't do that with the set type.

**Pull request merged**